nipype/interfaces/mrtrix3/preprocess.py

```diff
@@ -256,8 +256,9 @@ def _format_arg(self, name, trait_spec, value):
         return super()._format_arg(name, trait_spec, value)
 
     def _list_outputs(self):
-        outputs = self.output_spec().get()
-        outputs["out_file"] = op.abspath(self.inputs.out_file)
+        if self.inputs.out_file:
+            outputs = self.output_spec().get()
```

Comment:
L263 will fail if `self.inputs.bias`.

```suggestion
        outputs = self.output_spec().get()
        if self.inputs.out_file:
```